src/listener/nop_keyboard_listener.rs

```diff
@@ -0,0 +1,27 @@
+use std::error::Error;
+
+use super::common;
+use common::KeyboardListenerProvider;
+
+pub struct NopKeyboardListener {}
+
+impl KeyboardListenerProvider for NopKeyboardListener {
+    fn new(_callback: fn()) -> Result<NopKeyboardListener, Box<dyn Error>> {
+        return Ok(NopKeyboardListener {});
+    }
+
+    fn start(&mut self) -> Result<&str, Box<dyn Error>> {
+        println!(
```

Comment:
if you chose to implement the please panic on this at least

Reply:
this will make sure no one will think this is working for some reason.
the system should fail at this point